grpcreflect.go

```diff
@@ -284,6 +284,17 @@ type Namer interface {
 	Names() []string
 }
 
+// The NamerFunc type is an adapter to allow the use of ordinary functions as Namer,
+// it's useful when you want to use a function as a Namer but don't want to define a new type.
+// example: reflector := grpcreflect.NewReflector(grpcreflect.NamerFunc(func() []string { return s.names }))
```

Comment:
I think we can tweak the language a little and use a proper example block so it looks nicer when formatted in pkg.go.dev.
```suggestion
// NamerFunc is an adapter to allow the use of an ordinary function as a Namer.
// Example:
//
//	reflector := grpcreflect.NewReflector(grpcreflect.NamerFunc(
//		func() []string { return s.names },
//	))
```